os/drivers/usbhost/uvc/usbhost_uvc_ctrl.c

```diff
@@ -0,0 +1,1357 @@
+	}
+}
+
+/****************************************************************************
+ * Name: uvc_entity_match_guid
+ *
+ * Description:
+ *   This function compare the GUID of the entity and pass GUID, and return the
+ *   status.
+ *
+ * Input Parameters:
+ *   entity - pointer to entity object.
+ *   guid - guid array.
+ *
+ * Returned Values:
+ *   0 if matches, < 0 if entity guid is less than guid, otherwise > 0
+ *
+ ****************************************************************************/
+static int uvc_entity_match_guid(const struct uvc_entity *entity, const uint8_t guid[16])
+{
+	switch (UVC_ENTITY_TYPE(entity)) {
+	case UVC_ITT_CAMERA:
+		return memcmp(camera_guid, guid, 16) == 0;
+
+	case UVC_ITT_MEDIA_TRANSPORT_INPUT:
+		return memcmp(media_transport_input_guid, guid, 16) == 0;
+
+	case UVC_VC_PROCESSING_UNIT:
+		return memcmp(pu_guid, guid, 16) == 0;
+
+	case UVC_VC_EXTENSION_UNIT:
+		return memcmp(entity->extension.guidExtensionCode, guid, 16) == 0;
+
+	default:
+		return 0;
+	}
+}
+
+/****************************************************************************
+ * Name: uvc_find_control
+ *
+ * Description:
+ *   This function is used to get the individual controls of a unit
+ *
+ *   This function is called from the multiple methods to find appropriate
+ *   controls for a particular unit.
+ *
+ * Input Parameters:
+ *   entity - reference to unit
+ *   v4l2_id - v4l2 control id
+ *   v4l2_class - v4l2 class id
+ *   mapping  - pointer to hold mapping object.
+ *   control  - pointer to hold the control object
+ *   next - flag to check the next control in unit.
+ *
+ * Returned Values:
+ *   OK on success, appropriate error code on failures
+ *
+ ****************************************************************************/
+static void uvc_find_control(struct uvc_entity *entity, uint32_t v4l2_id, uint32_t v4l2_class, struct uvc_control_mapping **mapping, struct uvc_control **control, int next)
+{
+	struct uvc_control *ctrl = NULL;
+	struct uvc_control_mapping *map = NULL;
+	unsigned int i;
+
+	if (entity == NULL) {
+		udbg("Entity cannot be NULL\n");
+		return;
+	}
+
+	/* Each entity would holds multiple controls, retrieve each of them and
+	 * check if it matches with the parameters passed
+	 */
+	for (i = 0; i < entity->ncontrols; ++i) {
+		ctrl = &entity->controls[i];
+		if (!CTRL_IS_SUPPORTED(ctrl->status)) {
+			continue;
+		}
+		/* Parse the mapping list for the control */
+		map = queue_entry(ctrl->info.mappings.head, struct uvc_control_mapping, list);
+		for (; &map->list != (dq_entry_t *)&ctrl->info.mappings; map = queue_entry(map->list.flink, struct uvc_control_mapping, list)) {
+			if ((map->id == v4l2_id) && (map->class == v4l2_class) && !next) {
+				*control = ctrl;
+				*mapping = map;
+				return;
+			}
+
+			if ((*mapping == NULL || (*mapping)->id > map->id) && (map->id > v4l2_id) && next) {
+				*control = ctrl;
+				*mapping = map;
+			}
+		}
+	}
```

Comment:
Yes, it could happen, in such cases the queue will get empty and for loop exits.  So there is no explicit return requires, control moves to end of function.